src/components/root/abstractRecurringComponent.js

```diff
@@ -420,7 +420,8 @@ export default class AbstractRecurringComponent extends AbstractComponent {
 	 * @returns {boolean}
 	 */
 	canCreateRecurrenceExceptions() {
-		return this.isRecurring() || this.modifiesFuture()
+		const primaryIsRecurring = this.primaryItem?.isRecurring() ?? false
```

Comment:
can we actually use the null coalescence operator in this lib? https://github.com/nextcloud/nextcloud-vue/pull/2101 doesn't like it

Reply:
Seems like we can't. I just checked calendar and there we have an ignore exception for calendar js so that the library code gets transpiled. However, transpiling calendar-js directly will make maintaining harder because we have to commit artifacts all the time (not yet on npm).

So we have 2 options:
1. Transpile and publish calendar-js on npm.
2. Add the exception to webpack.common.js of nextcloud-vue (see below).

I'd prefer option 2 for the moment.

Adding `calendar-js` to `BabelLoaderExcludeNodeModulesExcept` in `nextcloud-vue/webpack.common.js` should fix it:
```
			{
				test: /\.js$/,
				loader: 'babel-loader',
				exclude: BabelLoaderExcludeNodeModulesExcept([
					'tributejs',
                                        'calendar-js',
				]),
			},
```
https://github.com/nextcloud/nextcloud-vue/blob/f92efdc4c5a731db19d71b71c342d3e6d15e1366/webpack.common.js#L140

Reply:
Ref https://github.com/nextcloud/nextcloud-vue/pull/2110

Reply:
> However, transpiling calendar-js directly will make maintaining harder because we have to commit artifacts all the time (not yet on npm).

Do we actually? Other libraries are also transpiled, e.g. from typescript to js. The artifacts are only created before the publish, they are *not* in git

Reply:
That would be great actually. Which library are you referring to? I had cdav-library in mind where we have to commit artifacts all the time.

Reply:
Right. so what's special about cdav-library and calendar-js is that we require them via a git(hub) package. We have to switch to the npm registry https://github.com/nextcloud/cdav-library/issues/400 and then this will work

https://github.com/nextcloud/nextcloud-axios is one of the transpiled ones, though it doesn't use rollup yet

Reply:
I heard about rollup being more suited for libraries because of tree-shaking. We should definitely look into it in the long run.

I also opened an issue here to track the publishing: https://github.com/nextcloud/calendar-js/issues/272